lib/src/update_data.dart

```diff
@@ -1,132 +1,176 @@
+  const UpdateInfo({
+    this.lastVersionAvailable,
+    required this.installedVersion,
+    this.requiredVersion,
+  });
+
+  /// Last version of the application.
+  final Version? lastVersionAvailable;
+
+  /// Current installed application version.
+  final Version installedVersion;
+
+  /// Minimum required version of the application.
+  final Version? requiredVersion;
+
+  factory UpdateInfo._fromMap(Map<dynamic, dynamic> map) {
+    final rawLatestVersionAvailable = map[Constants.lastVersionAvailable];
+    final rawInstalledVersion = map[Constants.installedVersion];
+    final rawRequiredVersion = map[Constants.requiredVersion];
+
+    return UpdateInfo(
+      lastVersionAvailable: rawLatestVersionAvailable != null
+          ? Version._fromMap(rawLatestVersionAvailable)
+          : null,
+      installedVersion: Version._fromMap(rawInstalledVersion),
+      requiredVersion: rawRequiredVersion != null
+          ? Version._fromMap(rawRequiredVersion)
+          : null,
+    );
   }
 }
 
-/// Version of the app. On Android only major is used.
+/// Application Version information.
+@immutable
 class Version {
-  /// App major version number. iOS and Android.
-  int major;
-
-  /// App minor version number. iOS only.
-  int minor;
-
-  /// App patch version number. iOS only.
-  int patch;
-
-  /// App build version number. iOS only.
-  int build;
-
-  static Version fromMap(Map<dynamic, dynamic> map) {
-    final Version version = Version();
-    version.major = map[Constants.major];
-    version.minor = map[Constants.minor];
-    version.patch = map[Constants.patch];
-    version.build = map[Constants.build];
-    return version;
+  const Version({
+    required this.major,
+    this.minor,
+    this.patch,
+    this.build,
+  });
+
+  /// Application major version number. iOS and Android.
+  final int major;
+
+  /// Application minor version number. iOS only.
+  final int? minor;
+
+  /// Application patch version number. iOS only.
+  final int? patch;
+
+  /// Application build version number. iOS only.
+  final int? build;
+
+  factory Version._fromMap(Map<dynamic, dynamic> map) {
+    return Version(
+      major: map[Constants.major] as int,
+      minor: map[Constants.minor] as int?,
+      patch: map[Constants.patch] as int?,
+      build: map[Constants.build] as int?,
+    );
   }
 
   String toString() {
-    return '$major.$minor.$patch';
-  }
-}
-
-class Metadata {
-  Map<String, dynamic> metadata;
-
-  static Metadata fromMap(Map<dynamic, dynamic> map) {
-    final Metadata metadata = Metadata();
-    metadata.metadata = {};
-    map.forEach((key, value) => metadata.metadata[key] = value);
-
-    return metadata;
+    final buffer = StringBuffer();
+    buffer.write(major);
+    if (minor != null) {
+      buffer.write('.$minor');
+    }
+    if (patch != null) {
+      buffer.write('.$patch');
+    }
+    if (build != null) {
+      buffer.write(' build:$build');
+    }
+    return buffer.toString();
```

Comment:
it might be easier to do: 
```dart
final version = [major, minor, patch].whereType<String>().join('.');
return [version, build].whereType<String>().join(':');
```
😄